src/pid.cpp

```diff
@@ -66,133 +76,131 @@ Pid::Pid(const Pid &source)
-
-  // Load integral clamp from param server or default to 0
-  double i_clamp;
-  nh.param("i_clamp", i_clamp, 0.0);
-  gains.i_max_ = std::abs(i_clamp);
-  gains.i_min_ = -std::abs(i_clamp);
-  if(nh.hasParam("i_clamp_min"))
-  {
-    nh.param("i_clamp_min", gains.i_min_, gains.i_min_); // use i_clamp_min parameter, otherwise keep -i_clamp
-    gains.i_min_ = -std::abs(gains.i_min_); // make sure the value is <= 0
-  }
-  if(nh.hasParam("i_clamp_max"))
-  {
-    nh.param("i_clamp_max", gains.i_max_, gains.i_max_); // use i_clamp_max parameter, otherwise keep i_clamp
-    gains.i_max_ = std::abs(gains.i_max_); // make sure the value is >= 0
-  }
-  nh.param("antiwindup", gains.antiwindup_, false);
-
-  nh.param("publish_state", publish_state_, false);
-
-  if(publish_state_)
-  {
-    state_publisher_.reset(new realtime_tools::RealtimePublisher<control_msgs::PidState>());
-    state_publisher_->init(nh, "state", 1);
-  }
-
-  setGains(gains);
-
-  reset();
-  initDynamicReconfig(nh);
 
-  return true;
+  setParameterEventCallback();
 }
 
-bool Pid::initXml(TiXmlElement *config)
+void Pid::initPid(double p, double i, double d, double i_max, double i_min, bool antiwindup)
 {
-  // Create node handle for dynamic reconfigure
-  ros::NodeHandle nh(DEFAULT_NAMESPACE);
-
-  double i_clamp;
-  i_clamp = config->Attribute("iClamp") ? atof(config->Attribute("iClamp")) : 0.0;
-
-  setGains(
-    config->Attribute("p") ? atof(config->Attribute("p")) : 0.0,
-    config->Attribute("i") ? atof(config->Attribute("i")) : 0.0,
-    config->Attribute("d") ? atof(config->Attribute("d")) : 0.0,
-    std::abs(i_clamp),
-    -std::abs(i_clamp),
-    config->Attribute("antiwindup") ? atof(config->Attribute("antiwindup")) : false
-  );
+  setGains(p, i, d, i_max, i_min, antiwindup);
 
   reset();
-  initDynamicReconfig(nh);
-
-  return true;
 }
 
-void Pid::initDynamicReconfig(ros::NodeHandle &node)
-{
-  ROS_DEBUG_STREAM_NAMED("pid","Initializing dynamic reconfigure in namespace "
-    << node.getNamespace());
-
-  // Start dynamic reconfigure server
-  param_reconfig_server_.reset(new DynamicReconfigServer(param_reconfig_mutex_, node));
-  dynamic_reconfig_initialized_ = true;
-
-  // Set Dynamic Reconfigure's gains to Pid's values
-  updateDynamicReconfig();
-
-  // Set callback
-  param_reconfig_callback_ = boost::bind(&Pid::dynamicReconfigCallback, this, _1, _2);
-  param_reconfig_server_->setCallback(param_reconfig_callback_);
-}
+// bool Pid::initParam(const std::string& prefix, const bool quiet)
+// {
+//   ros::NodeHandle nh(prefix);
+//   return init(nh, quiet);
+// }
+
+// bool Pid::init(const ros::NodeHandle &node, const bool quiet)
+// {
+//   ros::NodeHandle nh(node);
+//
+//   Gains gains;
+//
+//   // Load PID gains from parameter server
+//   if (!nh.getParam("p", gains.p_gain_))
+//   {
+//     if (!quiet) {
+//       ROS_ERROR("No p gain specified for pid.  Namespace: %s", nh.getNamespace().c_str());
+//     }
+//     return false;
```

Comment:
so much commented code, let's decide on it's fate

Reply:
imho, It's kind of redundant given that the parameters are already handled in other functions. I have preemptively removed it.